Jenkinsfile

```diff
@@ -1,34 +1,82 @@
+  [os: "fedora", ver: "39",      compiler: "gcc-13",  fairsoft: "jan24_patches",    image: "jan24"],
+  [os: "fedora", ver: "39",      compiler: "gcc-13",  fairsoft: "jan24_patches_mt", image: "jan24"],
+  [os: "ubuntu", ver: "20.04",   compiler: "gcc-9",   fairsoft: "jan24_patches",    image: "jan24"],
+  [os: "ubuntu", ver: "20.04",   compiler: "gcc-9",   fairsoft: "jan24_patches_mt", image: "jan24"],
+  [os: "ubuntu", ver: "20.04",   compiler: "gcc-9",   fairsoft: "nov22_patches",    image: "nov22"],
+  [os: "ubuntu", ver: "20.04",   compiler: "gcc-9",   fairsoft: "apr22_patches",    image: "apr22"],
+  [os: "ubuntu", ver: "22.04",   compiler: "gcc-11",  fairsoft: "jan24_patches",    image: "jan24"],
+  [os: "ubuntu", ver: "22.04",   compiler: "gcc-11",  fairsoft: "jan24_patches_mt", image: "jan24"],
+  [os: "ubuntu", ver: "22.04",   compiler: "gcc-11",  fairsoft: "nov22_patches",    image: "nov22"],
+  [os: "ubuntu", ver: "22.04",   compiler: "gcc-11",  fairsoft: "apr22_patches",    image: "apr22"],
+  [os: "ubuntu", ver: "rolling", compiler: "current", fairsoft: "dev", check: "warnings",
+    extra: "-DUSE_CLANG_TIDY=ON -DBUILD_MBS=OFF -DBUILD_PROOF_SUPPORT=OFF"],
+]
+
+def linux_checks = [
+  [os: 'ubuntu', ver: 'rolling', compiler: 'current', fairsoft: 'dev', check: 'doxygen'],
+]
+
+if (env.CHANGE_ID != null) { // only run checks for PRs
+  linux_checks += [
+    [os: "ubuntu", ver: "rolling", compiler: "current", fairsoft: "dev",  check: "format"]
+  ]
+}
+
+def jobMatrix(String type, String sourcedir, List specs) {
   def nodes = [:]
   for (spec in specs) {
-    def job = ''
+    def arch = spec.getOrDefault("arch", "x86_64")
+    def check = spec.getOrDefault("check", null)
+    def extra = spec.getOrDefault("extra", null)
+    def image = spec.getOrDefault("image", null)
+    def fairsoft = spec.fairsoft
+    def os = spec.os
+    def ver = spec.ver
+    def compiler = spec.compiler
+    def selector = "${os}-${ver}-${arch}"
+    def label = ""
     switch(type) {
-      case 'build':
-        job = "${spec.os}-${spec.ver}-${spec.arch}-${spec.compiler}-fairsoft-${spec.fairsoft}"
+      case "build":
+        label = "${os}-${ver}-${arch} ${compiler} ${fairsoft}"
         break
-      case 'check':
-        job = spec.check
+      case "check":
+        label = "Check ${check}"
         break
     }
-    def label = "${type}/${job}"
-    def selector = "${spec.os}-${spec.ver}-${spec.arch}"
-    def fairsoft = spec.fairsoft
-    def os = spec.os
-    def ver = spec.ver
-    def arch = spec.arch
-    def check = spec.getOrDefault("check", null)
-    def extra = spec.getOrDefault("extra", null)
-    def image = spec.getOrDefault("image", null)
 
     nodes[label] = {
       node(selector) {
-        githubNotify(context: "${prefix}/${label}", description: 'Building ...', status: 'PENDING')
+        githubNotify(context: "${label}", description: "Building ...", status: "PENDING")
         def logpattern = 'build/Testing/Temporary/*.log'
```

Comment:
Is the `logpattern` still correct this way?

Reply:
good point